Services/CO.CDP.EntityVerification/Ppon/OrganisationRegisteredEventHandler.cs

```diff
@@ -1,21 +1,35 @@
 using CO.CDP.EntityVerification.Events;
 using CO.CDP.EntityVerification.Persistence;
 using CO.CDP.MQ;
-using Microsoft.AspNetCore.Components;
-using System.Transactions;
+using Identifier = CO.CDP.EntityVerification.Persistence.Identifier;
 
 namespace CO.CDP.EntityVerification.Ppon;
 
-public class OrganisationRegisteredEventHandler(IPponService pponService, IPponRepository pponRepository, IPublisher publisher)
+public class OrganisationRegisteredEventHandler(IPponService pponService,
+    IPponRepository pponRepository,
+    IPublisher publisher)
     : IEventHandler<OrganisationRegistered>
 {
     public async Task Handle(OrganisationRegistered @event)
     {
-        var pponId = pponService.GeneratePponId();
+        Persistence.Ppon newPpon = new() {
+            PponId = pponService.GeneratePponId(),
+            Name = @event.Name,
+            OrganisationId = @event.Id
+        };
 
-        Persistence.Ppon newIdentifier = new() { PponId = pponId };
+        List<Identifier> ids = [];
+        foreach (var e in @event.AllIdentifiers())
+        {
+            ids.Add(new Identifier { Ppon = newPpon,
+                LegalName = e.LegalName,
+                Scheme = e.Scheme,
+                Uri = e.Uri });
+        }
```

Comment:
If we hide this  mappinglogic behind a private method, the call could just be made in the field initialiser, i.e.

```csharp
Ppon newPpon = new () {
    // ...
    Identifiers = asPerisistenceIdentifiers(@event.AllIdentifiers())
}
```

This way, side effects (adding to a mutable list) would also be isolated.